src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,249 @@
+    /**
+     * List containing either a notification that WiFi is disabled, or the list of saved
+     * networks.
+     */
+    private ListView mListWifi;
+    /**
+     * The configured WiFi networks, or null, if WiFi is not available.
+     */
+    private WifiConfiguration[] mConfiguredNetworks;
+    /**
+     * The SSID of the networks that have been enabled (in selective mode)
+     */
+    private Set<String> mSelectedSsids;
+    /**
+     * Adapter between the data above and the ListView
+     */
+    private WifiListAdapter listAdapter;
+
+    public WifiSsidPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.preference_wifi_ssids);
+    }
+
+    @Override
+    protected View onCreateDialogView() {
+        View view = super.onCreateDialogView();
+        mListWifi = (ListView) view.findViewById(R.id.listWifi);
+        return view;
+    }
+
+    /**
+     * This is where the data is actually bound.
+     */
+    @Override
+    protected void onBindDialogView(View view) {
+        super.onBindDialogView(view);
+
+        if (isPersistent()) {
+            // currently persisted values are pre-selected in the list
+            String key = getKey();
+            Set<String> ssids = getSharedPreferences().getStringSet(key, null);
+            mSelectedSsids = ssids != null ? new HashSet<>(ssids) : new HashSet<String>();
+        } else {
+            mSelectedSsids = new HashSet<>(); // if non-persistent, nothing is pre-selected
+        }
+
+        mConfiguredNetworks = loadConfiguredNetworksSorted();
+        listAdapter = new WifiListAdapter();
+        mListWifi.setAdapter(listAdapter);
+    }
+
+    /**
+     * Load the configured WiFi networks, sort them by SSID.
+     *
+     * @return a sorted array of WifiConfiguration, or null, if data cannot be retrieved
+     */
+    private WifiConfiguration[] loadConfiguredNetworksSorted() {
+        WifiManager wifiManager = (WifiManager) getContext().getSystemService(Context.WIFI_SERVICE);
+        if (wifiManager != null) {
+            List<WifiConfiguration> configuredNetworks = wifiManager.getConfiguredNetworks();
+            // if WiFi is turned off, getConfiguredNetworks returns null on many devices
+            if (configuredNetworks != null) {
+                WifiConfiguration[] result = configuredNetworks.toArray(new WifiConfiguration[configuredNetworks.size()]);
+                Arrays.sort(result, new Comparator<WifiConfiguration>() {
+                    @Override
+                    public int compare(WifiConfiguration lhs, WifiConfiguration rhs) {
+                        return lhs.SSID.compareToIgnoreCase(rhs.SSID);
+                    }
+                });
+                return result;
+            }
+        }
+        // WiFi is turned of or device doesn't have WiFi
+        return null;
+    }
+
+    @Override
+    protected void showDialog(Bundle state) {
+        super.showDialog(state);
```

Comment:
Not entirely sure what you mean, if I remove the call to super-class the dialog won't ever be shown. If I just conditionally don't show it and show a Toast instead, I'll need to check for side-effects. Will have a look into that.